newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,301 @@
+      console.error('Dialog was opened for an unsupported event text color.');
+      return '';
+    }
+  };
+
+  getBackgroundColors = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let backgroundColors;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      backgroundColors = {
+        red: commentEvent.getBackgroundColorRed(),
+        green: commentEvent.getBackgroundColorGreen(),
+        blue: commentEvent.getBackgroundColorBlue(),
+      };
+
+      return backgroundColors;
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      const groupEvent = gd.asGroupEvent(event);
+      backgroundColors = {
+        red: groupEvent.getBackgroundColorR(),
+        green: groupEvent.getBackgroundColorG(),
+        blue: groupEvent.getBackgroundColorB(),
+      };
+
+      return backgroundColors;
+    } else {
+      console.error(
+        'Dialog was opened for an unsupported event background color'
+      );
+      return '';
+    }
+  };
+
+  state = {
+    textValue: this.getComment(),
+    textColors: this.getTextColors(),
+    backgroundColors: this.getBackgroundColors(),
+  };
+
+  render() {
+    const { event, onApply, onClose } = this.props;
+    const eventType = event.getType();
+
+    return (
+      <Dialog
+        title={<Trans>Text editor</Trans>}
+        onRequestClose={onClose}
+        open
+        noMargin
+        actions={[
+          <FlatButton
+            key="close"
+            label={<Trans>Cancel</Trans>}
+            primary={false}
+            onClick={onClose}
+          />,
+          <FlatButton
+            key={'Apply'}
+            label={<Trans>Apply</Trans>}
+            primary
+            keyboardFocused
+            onClick={() => {
+              if (eventType === 'BuiltinCommonInstructions::Comment') {
+                this.setComment();
+              } else if (eventType === 'BuiltinCommonInstructions::Group') {
+                this.setName();
+              }
+              this.setTextColors();
+              this.setBackgroundColors();
+
+              onApply();
+            }}
+          />,
+        ]}
+        secondaryActions={[
+          <HelpButton
+            key="help"
+            helpPagePath="/interface/scene-editor/layers-and-cameras"
+          />,
+        ]}
+      >
+        <Column noMargin>
+          <MiniToolbar>
+            <MiniToolbarText>
+              <Trans />
+            </MiniToolbarText>
+            <MiniToolbarText>
+              <Trans>Background color:</Trans>
+            </MiniToolbarText>
+            <ColorPicker
+              style={styles.sizeTextField}
+              disableAlpha
+              color={{
+                r: this.state.backgroundColors.red,
+                g: this.state.backgroundColors.green,
+                b: this.state.backgroundColors.blue,
+                a: 255,
+              }}
```

Comment:
Ideally, we could store in the state objects with the same shape as the ones passed to the ColorPicker:

colors={this.state.backgroundColor}